src/librustc_middle/query/mod.rs

```diff
@@ -711,6 +711,14 @@ rustc_queries! {
             desc { "destructure constant" }
         }
 
+        /// Dereference a constant reference or raw pointer and turn the result into a constant
+        /// again.
```

Comment:
Patterns compare references by the value they refer to, but raw pointers by their address. So why are both mentioned together here?

Reply:
Ah I see, pattern matching uses it just for references but it could also work for raw pointers. Makes sense.